src/dragonball/src/device_manager/mod.rs

```diff
@@ -1011,3 +1011,168 @@ impl DeviceManager {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::{Arc, Mutex};
+
+    use kvm_ioctls::Kvm;
+    use test_utils::skip_if_not_root;
+    use vm_memory::{GuestAddress, MmapRegion};
+
+    use super::*;
+    use crate::vm::CpuTopology;
+
+    impl DeviceManager {
+        pub fn new_test_mgr() -> Self {
+            let kvm = Kvm::new().unwrap();
+            let vm = kvm.create_vm().unwrap();
+            let vm_fd = Arc::new(vm);
+            let epoll_manager = EpollManager::default();
+            let res_manager = Arc::new(ResourceManager::new(None));
+            let logger = slog_scope::logger().new(slog::o!());
+
+            DeviceManager {
+                vm_fd: Arc::clone(&vm_fd),
+                con_manager: ConsoleManager::new(epoll_manager, &logger),
+                io_manager: Arc::new(ArcSwap::new(Arc::new(IoManager::new()))),
+                io_lock: Arc::new(Mutex::new(())),
+                irq_manager: Arc::new(KvmIrqManager::new(vm_fd.clone())),
+                res_manager,
+
+                legacy_manager: None,
+                #[cfg(feature = "virtio-blk")]
+                block_manager: BlockDeviceMgr::default(),
+                #[cfg(feature = "virtio-fs")]
+                fs_manager: Arc::new(Mutex::new(FsDeviceMgr::default())),
+                #[cfg(feature = "virtio-net")]
+                virtio_net_manager: VirtioNetDeviceMgr::default(),
+                #[cfg(feature = "virtio-vsock")]
+                vsock_manager: VsockDeviceMgr::default(),
+                #[cfg(target_arch = "aarch64")]
+                mmio_device_info: HashMap::new(),
+
+                logger,
+            }
+        }
+    }
+
+    #[test]
+    fn test_create_device_manager() {
+        skip_if_not_root!();
+        let mgr = DeviceManager::new_test_mgr();
+        let _ = mgr.io_manager();
+    }
+
+    #[cfg(target_arch = "x86_64")]
+    #[test]
+    fn test_create_legacy_device_manager() {
```

Comment:
is there other redundant unit test called `test_create_legacy_device_manager` under `/device_manager/legacy.rs`?
Should we remove it?

Reply:
The content of the two tests is different and has been renamed to test_create_devices.
Thanks.